src/qibocal/protocols/characterization/flux_dependence/resonator_flux_dependence.py

```
@@ -24,10 +25,72 @@ class ResonatorFluxParameters(Parameters):
     """Width for frequency sweep relative to the readout frequency [Hz]."""
     freq_step: int
     """Frequency step for sweep [Hz]."""
-    bias_width: float
+    bias_width: Optional[float] = None
     """Width for bias sweep [V]."""
-    bias_step: float
+    bias_step: Optional[float] = None
     """Bias step for sweep [a.u.]."""
+    flux_amplitude_start: Optional[Union[float, List[float]]] = None
+    """Amplitude start value(s) for flux pulses sweep relative to the qubit sweetspot [a.u.]."""
+    flux_amplitude_end: Optional[Union[float, List[float]]] = None
+    """Amplitude end value(s) for flux pulses sweep relative to the qubit sweetspot [a.u.]."""
+    flux_amplitude_step: Optional[Union[float, List[float]]] = None
+    """Amplitude step(s) for flux pulses sweep [a.u.]."""
+
+    def __post_init__(self):
+        if not self.has_bias_params:
+            if self.has_flux_params:
+                self.check_flux_params()
+                return
+        if not self.has_flux_params:
+            if self.has_bias_params:
+                return
+        raise ValueError(
+            "Too many arguments provided. Provide either bias_width "
+            "and bias_step or flux_amplitude_width and flux_amplitude_step."
+        )
+
+    def check_flux_params(self):
+        """All flux params must be either all float or all lists with the same length.
```

@andrea-pasquale should we check here for stuff like this or just let the experiment crash ?

---

Checking doesn't hurt, I would personally keep it.